Source/WebKit/WebProcess/WebPage/WebCookieJar.cpp

```diff
@@ -278,49 +280,82 @@ void WebCookieJar::deleteCookie(const WebCore::Document& document, const URL& ur
 
 void WebCookieJar::getCookiesAsync(WebCore::Document& document, const URL& url, const WebCore::CookieStoreGetOptions& options, CompletionHandler<void(std::optional<Vector<WebCore::Cookie>>&&)>&& completionHandler) const
 {
-    RefPtr webFrame = document.frame() ? WebFrame::fromCoreFrame(*document.frame()) : nullptr;
-    if (!webFrame || !webFrame->page()) {
+    auto frame = document.frame();
+    if (!frame) {
         completionHandler({ });
         return;
     }
 
+    std::optional<WebCore::FrameIdentifier> frameID = std::nullopt;
+    std::optional<WebCore::PageIdentifier> pageID = std::nullopt;
+
     ApplyTrackingPrevention applyTrackingPreventionInNetworkProcess = ApplyTrackingPrevention::No;
 #if ENABLE(TRACKING_PREVENTION)
-    if (shouldBlockCookies(webFrame.get(), document.firstPartyForCookies(), url, applyTrackingPreventionInNetworkProcess)) {
-        completionHandler({ });
-        return;
+    if (frame->loader().client().isRemoteWorkerFrameLoaderClient()) {
+        if (shouldBlockCookies(nullptr, document.firstPartyForCookies(), url, applyTrackingPreventionInNetworkProcess)) {
+            completionHandler({ });
+            return;
+        }
+    } else {
+        RefPtr webFrame = document.frame() ? WebFrame::fromCoreFrame(*document.frame()) : nullptr;
```

Comment:
you create a `frame` variable above.

Reply:
Do you mean that I don't need this at all and can just use the `frame` variable I already made in place of all the places I use this 'webFrame'? Are we using any properties that webFrame has but frame doesn't?

Reply:
Oh I misread the comment. You just mean that I can directly do `WebFrame::fromCoreFrame(*document.frame())`. Understood.

Reply:
You defined `Ref frame = document.frame()` earlier in this function so use it increase of repeatedly calling `document.frame()`.